File: firstspringdemo/src/main/java/firstspring/firstspringdemo/DTOs/InventoryDto.java
package firstspring.firstspringdemo.DTOs;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import java.util.UUID;

public class InventoryDto {

    private UUID inventoryId;

    @NotNull(message = "Product ID is required")
    private UUID productId;

    @NotNull(message = "Location ID is required")
    private UUID locationId;

    @NotNull(message = "Quantity is required")
    @Min(value = 0, message = "Quantity must be a non-negative number")
    private Integer quantity;

    private Integer reorderPoint; //Optional
    private Integer reorderQuantity; //Optional

    // Constructors (default, and with all fields)
    public InventoryDto() {}

    public InventoryDto(UUID inventoryId, UUID productId, UUID locationId, Integer quantity, Integer reorderPoint, Integer reorderQuantity) {
        this.inventoryId = inventoryId;
        this.productId = productId;
        this.locationId = locationId;
        this.quantity = quantity;
        this.reorderPoint = reorderPoint;
        this.reorderQuantity = reorderQuantity;
    }

    // Getters and setters
    public UUID getInventoryId() {
        return inventoryId;
    }

    public void setInventoryId(UUID inventoryId) {
        this.inventoryId = inventoryId;
    }

    public UUID getProductId() {
        return productId;
    }

    public void setProductId(UUID productId) {
        this.productId = productId;
    }

    public UUID getLocationId() {
        return locationId;
    }

    public void setLocationId(UUID locationId) {
        this.locationId = locationId;
    }

    public Integer getQuantity() {
        return quantity;
    }

    public void setQuantity(Integer quantity) {
        this.quantity = quantity;
    }

    public Integer getReorderPoint() {
        return reorderPoint;
    }

    public void setReorderPoint(Integer reorderPoint) {
        this.reorderPoint = reorderPoint;
    }

    public Integer getReorderQuantity() {
        return reorderQuantity;
    }

    public void setReorderQuantity(Integer reorderQuantity) {
        this.reorderQuantity = reorderQuantity;
    }
}

File: firstspringdemo/flutter/lib/models/inventory_dto.dart
class InventoryDto {
  final String? inventoryId;
  final String? productId;
  final String? locationId;
  final int? quantity;
  final int? reorderPoint;
  final int? reorderQuantity;

  InventoryDto({
    this.inventoryId,
    this.productId,
    this.locationId,
    this.quantity,
    this.reorderPoint,
    this.reorderQuantity,
  });

  factory InventoryDto.fromJson(Map<String, dynamic> json) {
    return InventoryDto(
      inventoryId: json['inventoryId'],
      productId: json['productId'],
      locationId: json['locationId'],
      quantity: json['quantity'],
      reorderPoint: json['reorderPoint'],
      reorderQuantity: json['reorderQuantity'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'inventoryId': inventoryId,
      'productId': productId,
      'locationId': locationId,
      'quantity': quantity,
      'reorderPoint': reorderPoint,
      'reorderQuantity': reorderQuantity,
    };
  }
}
